cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -242,7 +242,8 @@ public void setReducerResources(Resources resources) {
   }
 
   private static MetricsCollector getMetricCollector(MetricsCollectionService service, Program program,
-                                                     MapReduceMetrics.TaskType type, String runId, String taskId) {
+                                                     MapReduceMetrics.TaskType type, String runId, String taskId,
```

Comment:
not introduced by you, but some of these other arguments are nullable so we should be consistent and annotate them.